### ISSUE TITLE
Python 3 migration plan

### ISSUE THREAD

**Issue by JnyJny** [bwelton/diogenes#1]
Python 2.7 will be EOL in the very near term, do you have a plan for migrating your python source to version 3?

**Comment by bwelton** [comment]
Yeah this is actually on my list of things to do as part of a major cleanup operation that I am slowly working on right now to both make Diogenes easier to compile/run and to improve stability. 

In the future, Python may actually be dropped since its inclusion/usage was partly driven by some past experiments that I was working on to test out techniques to use in Diogenes. There is an absolutely enormous amount of dead code/unused code in this repo right now (from these past experiments) that needs to be removed  I am hopeful that I can get around to this shortly (after my current paper/my defense is complete).

**Comment by JnyJny** [comment]
Good luck with your defense :)

Some idea of what code is dead and what may live on would be helpful in driving future 2to3 porting efforts (if needed at all).

**Comment by bwelton** [comment]
Most of my current work has been in src/driverapi/ and src/driverapi/launcher (contains most of the instrumentation code) with the analysis directory containing the python scripts to generate the final output/visualization. 

This weekend I will do a quick write up as to what each file in these directories does/its purpose (likely posting a wiki entry on this). That should help guide you a bit on where to find specific things of interest. 

Again, sorry about the sort of cluttered state of this repo.
